assets/javascripts/validation/form.js

```diff
@@ -122,8 +122,12 @@ var flushHiddenElementErrors = function (invalidInputs) {
 
     if ($elem.is(':hidden')) {
       delete invalidInputs[inputName];
-      $elem.val('');
-      $elem.closest('.form-field-group').removeClass('form-field-group--error');
+
+      // filter out radios and checkboxes before clearing the element value   
+      if ($elem.filter(function(e){ return $(e).is('[type!="radio"]') && $elem.is('[type!="checkbox"]'); }).length) {
+        $elem.val('');
+        $elem.closest('.form-field-group').removeClass('form-field-group--error');
+      }
     }
   }
```

Comment:
@mattpepper for the conditional something like the following feels a little clearer to me, then you don't need the comment:

```
var inputType = $elem.prop('type');

if (inputType !== 'radio' && inputType !== 'checkbox') {
   $elem.val('');
   $elem.closest('.form-field-group').removeClass('form-field-group--error');
}
```

Can you also add a test around this for assets-frontend please in https://github.com/hmrc/assets-frontend/blob/master/assets/test/specs/form.spec.js